proposals/0123-block-fee-distribution.md

```diff
@@ -0,0 +1,145 @@
+---
+simd: '0123'
+title: Block Fee Distribution
+authors: Justin Starry (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-03-10
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+A new mechanism for distributing block fees to delegated stake accounts is
+proposed to allow validators to share block fee revenue with their delegators.
+Validators will be able to specify a commission rate for block fees and the
+protocol will automatically distribute the remaining fees to delegated stake
+accounts at the end of each epoch.
+
+## Motivation
+
+Delegated stake directly increases the number of blocks that a validator is allocated
+in an epoch leader schedule but the core protocol doesn't support diverting any of
+that extra revenue to stake delegators.
+
+## Alternatives Considered
+
+Due to the lack of core protocol support for this feature, validators have
+developed independent ways to distribute block revenue which is not enforced by
+the core protocol. For example, the Cogent validator diverts part of its fee
+revenue to NFT holders. But it's up the NFT holders to audit and hold Cogent
+accountable to promised commission.
+
+Another alternative is Jito's mechanism for block "tips" (not fees, but the idea
+is similar). Jito's validator implementation includes a tip distribution program
+which it instructs validator operators to divert all of their tips to but cannot
+enforce perfect compliance. It's up to stakers and the Jito team to audit
+compliance by validator operators. This mechanism requires trusting a
+third-party (in this case Jito) to calculate reward distribution in an accurate
+and fair mannger. It also relies on using a merkle tree to distribute fees to
+all stake accounts and the distributed fees are not automatically staked in
+recipient stake accounts.
+
+## New Terminology
+
+- validator fee program: New core program which supports diverting a portion of
+block fees to stake delegators
+
+- validator fee account: New account type which supports setting a commission
```

Comment:
Wdyt about having a separate SIMD just for this piece? This on its own is a valuable feature

Reply:
Yup, was already writing one up, here's a draft that you can check out. Would appreciate feedback from you and @buffalu. This SIMD now just focuses on 1) where to send commission from different revenue sources and 2) how to collect and distribute post-commission revenue to stake delegators

Reply:
oops SIMD draft here: https://github.com/solana-foundation/solana-improvement-documents/pull/185